src/api/api.ts

```diff
@@ -0,0 +1,251 @@
+  productId: number;
+  productOptionId: number;
+  productQuantity: number;
+  messageCardTemplateId: number;
+  messageCardTextMessage: string;
+  senderId: number;
+  receiverId: number;
+  hasCashReceipt: boolean;
+  cashReceiptType?: 'PERSONAL' | 'BUSINESS';
+  cashReceiptNumber?: string;
+}
+
+// api 호출 함수
+export const fetchRankingProducts = async (
+  params: GetRankingProductsRequest,
+): Promise<GetRankingProductsResponse> => {
+  const response = await axiosInstance.get<GetRankingProductsResponse>('/api/v1/ranking/products', {
+    params,
+  });
+  return response.data;
+};
+
+export const fetchThemes = async (): Promise<GetThemesResponse> => {
+  const response = await axiosInstance.get<GetThemesResponse>('/api/v1/themes');
+  return response.data;
+};
+
+export const fetchThemeProducts = async (
+  params: GetThemeProductsRequest,
+): Promise<GetThemeProductsResponse> => {
+  const { themeKey, ...queryParams } = params;
+  const response = await axiosInstance.get<GetThemeProductsResponse>(
+    `/api/v1/themes/${themeKey}/products`,
+    { params: queryParams },
+  );
+  return response.data;
+};
+
+export const fetchProductDetail = async (
+  params: GetProductDetailRequest,
+): Promise<GetProductDetailResponse> => {
+  const { productId } = params;
+  const response = await axiosInstance.get<GetProductDetailResponse>(
+    `/api/v1/products/${productId}/detail`,
+  );
+  return response.data;
+};
+
+export const fetchProductOptions = async (
+  params: GetProductOptionsRequest,
+): Promise<GetProductOptionsResponse> => {
+  const { productId } = params;
+  const response = await axiosInstance.get<GetProductOptionsResponse>(
+    `/api/v1/products/${productId}/options`,
+  );
+  return response.data;
+};
+
+export const fetchMessageCardTemplates = async (): Promise<GetMessageCardTemplatesResponse> => {
+  const response = await axiosInstance.get<GetMessageCardTemplatesResponse>(
+    '/api/v1/message-card/templates',
+  );
+  return response.data;
+};
+
+export const fetchMyAccountInfo = async (): Promise<GetMyAccountInfoResponse> => {
+  const response = await axiosInstance.get<GetMyAccountInfoResponse>('/api/v1/my-account/info');
+  return response.data;
+};
+
+export const fetchWishProducts = async (
+  params: GetWishProductsRequest,
+): Promise<GetWishProductsResponse> => {
+  const response = await axiosInstance.get<GetWishProductsResponse>(
+    '/api/v1/my-account/wish/products',
+    { params },
+  );
+  return response.data;
+};
+
+export const putMyAccountPoint = async (
+  data: PutMyAccountPointRequest,
+): Promise<PutMyAccountPointResponse> => {
+  const response = await axiosInstance.put<PutMyAccountPointResponse>(
+    '/api/v1/my-account/point',
+    data,
+  );
+  return response.data;
+};
+
+export const postOrder = async (data: PostOrderRequest): Promise<PostOrderResponse> => {
+  try {
+    const response = await axiosInstance.post<PostOrderResponse>('/api/v1/order', data);
```

Comment:
```suggestion
export const postOrder = async (data: PostOrderRequest): Promise<void> => {
  try {
    const response = await axiosInstance.post<void>('/api/v1/order', data);
```

void 타입은 따로 타입 만들기보다는 직접 inline 하셔도 괜찮습니다.

Reply:
step3 브랜치에서 반영하였습니다.